### DISCUSSION TITLE
How to reset all filters and show all data using PhpSpreadsheet?

### DISCUSSION THREAD

**Discussion by modestguy** [PHPOffice/PhpSpreadsheet#3367] [Q&A] [answered]
Hi, guys! Please help me. I have an xlsx document with filter in one column. Without filter i have about 100 records. With filter it shows 2 records. I load document using phpspreadsheet and want to remove any filters and show all 100 records. How to do it?

I try:
`$dataSheet->removeAutoFilter();` but it show only 2 records.
Then try  `$dataSheet->setAutoFilter(null);` - show only 2 records.
Try to add ` $autoFilter->showHideRows();` but nothing happens.
Please, help me if you know how to do it. Thanks for answers! Sorry for my english.

**Comment by MarkBaker** [comment]
Is this AutoFilter in the datasheet; or is it in a Table in the DataSheet?

Does a call to `$datasheet->getAutoFilter()` return an object?
Does a call to `$datasheet->getTableCollection()` return an ArrayObject that contains one or more Table objects?

**Reply by modestguy** [reply]
@MarkBaker in debug mode:
[autoFilter object](https://i.ibb.co/4mTWhNM/auto-Filter-Object.png)
[tableCollection](https://i.ibb.co/qNdDf40/table-Collection.png)

**Comment by MarkBaker** [comment, marked as answer]
Ok! So it's just a standard AutoFilter

You'll need to reset the filter rules for each column before you call `showHideRows()`
```php
$autoFilter = $dataSheet->getAutoFilter();
foreach($autoFilter->getColumns() as $column) {
    $autoFilter->clearColumn($column->getColumnIndex());
}
$autoFilter->showHideRows();
```

Then, if you still want to remove the AutoFilter, you can do so
```php
$dataSheet->removeAutoFilter();
```

**Reply by modestguy** [reply]
Thank you very much!  God bless u! 
It's solution of my problem.
The question is closed, subscribed to you on linkedIn.
